tests/table.spec.ts

```diff
@@ -128,71 +130,182 @@ test.describe('tests complex table operations', () => {
 
     // Check snapshot
     await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
 
+    // Clear search query
     await page.keyboard.press('Backspace');
 
+    await expect(searchBar).toHaveValue('');
+
+    // Wait until it's done loading
+    await expect(
+      page.locator('.iris-grid .iris-grid-loading-status')
+    ).toHaveCount(0);
+
     // Check snapshot
     await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
   });
+
+  test('can organize columns', async () => {
+    await page.locator('data-testid=menu-item-Organize Columns').click();
+
+    await test.step('Search', async () => {
+      await page.getByPlaceholder('Search').click();
+      await page.keyboard.type('dou');
+
+      await expect(
+        page.locator('.visibility-ordering-builder')
+      ).toHaveScreenshot();
+    });
+
+    await test.step('Move Selection Down', async () => {
+      await page
+        .getByRole('button', { name: 'Move selection down' })
+        .click({ clickCount: 2 });
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+    });
+
+    await test.step('Move Selection Up', async () => {
+      await page.getByRole('button', { name: 'Move selection up' }).click();
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+    });
+
+    await test.step('Move Selection to Bottom', async () => {
+      await page
+        .getByRole('button', { name: 'Move selection to bottom' })
+        .click();
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+    });
+
+    await test.step('Move Selection to Top', async () => {
+      await page.getByRole('button', { name: 'Move selection to top' }).click();
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+    });
+
+    await test.step('Sort Descending', async () => {
+      await page.getByRole('button', { name: 'Sort descending' }).click();
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+    });
+
+    await test.step('Sort Ascending', async () => {
+      await page.getByRole('button', { name: 'Sort ascending' }).click();
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+    });
+
+    await test.step('Hide Selected', async () => {
+      await page.getByRole('button', { name: 'Hide Selected' }).click();
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+    });
+
+    await test.step('Reset', async () => {
+      await page.getByRole('button', { name: 'Reset' }).click();
+
+      await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+    });
+
+    await test.step('Drag', async () => {
+      const dragColumnButton = page.getByRole('button', {
+        name: 'Toggle visibility Float',
+      });
+      const [x, y] = await dragColumnButton
+        .boundingBox()
+        .then(pos => (pos && pos.x && pos.y ? [pos.x, pos.y - 50] : [0, 0]));
```

Comment:
You're checking a number here, should be checking `pos.x != null` instead of just `pos.x`, as that will fail if `x` is zero (which is unlikely in this case, but should always be explicit when checking numbers).